### PR TITLE
Pallets: Asset Rate - Rename AssetId to AssetKind, Introduce AssetKindFactory Trait

### PR DESCRIPTION
**Pull request by muharem**
In the Asset Rate Pallet, the `AssetId` type parameter has been renamed to `AssetKind`. Alongside this change, the `AssetKindFactory` trait has been introduced specifically for benchmarks.

For Polkadot and Kusama, the `AssetKind` will be represented as a type consisting of `(xcm::MultiLocation, xcm::AssetId)`. This renaming is done to avoid naming collisions.  And since it is not suitable to implement `From<u32>` for such a type, we introduce the `AssetKindFactory` trait as a solution.

### CODE REVIEW

**Review by joepetrowski — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3230766

File: frame/asset-rate/Cargo.toml [hunk 1]
```diff
@@ -22,10 +22,10 @@ frame-support = { version = "4.0.0-dev", default-features = false, path = "../su
 frame-system = { version = "4.0.0-dev", default-features = false, path = "../system" }
 sp-runtime = { version = "24.0.0", default-features = false, path = "../../primitives/runtime" }
 sp-std = { version = "8.0.0", default-features = false, path = "../../primitives/std" }
+sp-core = { version = "21.0.0", default-features = false, path = "../../primitives/core" }
```
Comment (ggwpez) [inline frame/asset-rate/Cargo.toml:25]: Could be `optional` if its only used for benchmarking.

**Review by ggwpez — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3266281

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3266526

**Comment by muharem** [comment]
bot merge

**Pull request merged**